lib/tree.rb

```diff
@@ -1,38 +1,61 @@
 class NoApplesError < StandardError; end
 
 class Tree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_reader :height, :age, :apples, :alive
 
-  def initialize
+  def initialize #constructor to initalize objects
+    @height = 1
+    @age = 1
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1 
+    @height += 1
+    @alive = false if @age > 15 
+    add_apples if @age > 3
   end
 
   def add_apples
+    colors = ["red","green"]
+    diameter = rand(1..10)
+    number_of_apples = rand(10..20)
+    number_of_apples.times do 
+      apple = Apple.new(colors.sample,diameter)
+      @apples << apple
+    end
   end
 
   def any_apples?
+    apples.length > 0
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    apple = apples.last
```

Comment:
I'm pretty sure `@apples.pop` returns this last apple while removing it from the array.